src/repositories/ContractEntityRepository.ts

```diff
@@ -58,18 +58,30 @@ export default class ContractEntityRepository extends Subject<ContractEntity> {
           primaryKey: true,
         },
         entityType: {
-          type: DataTypes.ENUM({
-            values: [
+          primaryKey: true,
+          type: DataTypes.ENUM,
+          values: [
               EntityType.StakeIntentDeclareds,
               EntityType.StakeRequesteds,
               EntityType.StateRootAvailables,
               EntityType.StakeIntentConfirmeds,
               EntityType.StakeProgresseds,
               EntityType.MintProgresseds,
               EntityType.GatewayProvens,
-            ],
-          }),
-          primaryKey: true,
+          ],
+          validate: {
```

Comment:
Do we really need this validation? I mean, we've already declared it to be enum and supplied the values for that enumeration. It's like checking the type ... 

I've quickly googled. Please, have a look into this two links:
  - https://github.com/sequelize/sequelize/issues/5765
  - https://github.com/sequelize/sequelize/issues/8965

One proposal is to try to create enum like this:

```ts
type: Sequelize.ENUM(EntityType.StakeIntentDeclareds, ...)
```

And if it's not fixed, we can try this simpler validation mechanism:

```ts
isIn: {
  args: [[ value-goes-here ]],
  msg: 'Invalid entity type is specified.'
}
```

Reply:
Thanks.
I tried the proposal, but got error `Property 'ENUM' does not exist on type 'typeof Sequelize'.`

Passing `typeValidation: true` to sequelize constructor enables validation of ENUM values.